coderd/database/queries/workspaces.sql

```diff
@@ -345,6 +347,27 @@ WHERE
 			  (latest_build.template_version_id = template.active_version_id) = sqlc.narg('using_active') :: boolean
 		  ELSE true
 	END
+	-- Filter by has_ai_task in latest build
+	AND CASE
+		WHEN sqlc.narg('has_ai_task') :: boolean IS NOT NULL THEN
+			(COALESCE(latest_build.has_ai_task, false) OR (
+				-- If the build has no AI task, it means that the provisioner job is in progress
+				-- and we don't know if it has an AI task yet. In this case, we optimistically
+				-- assume that it has an AI task if the AI Prompt parameter is not empty. This
+				-- lets the AI Task frontend spawn a task and see it immediately after instead of
+				-- having to wait for the build to complete.
+				latest_build.has_ai_task IS NULL AND
+				latest_build.completed_at IS NULL AND
+				EXISTS (
+					SELECT 1
+					FROM workspace_build_parameters
+					WHERE workspace_build_parameters.workspace_build_id = latest_build.id
+					AND workspace_build_parameters.name = 'AI Prompt'
```

Comment:
Hmm we may want to accept this name as an arg, but could get tricky since this is used in a few places.

Reply:
If we want to enforce that this constant matches the constant in go code, I'd rather define a helper function:

```sql
CREATE FUNCTION ai_prompt_param_name() RETURNS TEXT AS $$
BEGIN
    RETURN 'AI Prompt';
END;
$$ LANGUAGE plpgsql IMMUTABLE;
```

Then use that function in the query, and add a test that checks if the result of that function matches the constant in go code.